src/Ast.ml

```diff
@@ -106,24 +106,12 @@ let index_op_get_sugar (ident : Longident.t) args =
       | [{pexp_desc= Pexp_array l}] when List.length l > 3 ->
           Some (index_op_bigarray, l)
       | _ -> None )
-    | _ -> None
+    | _ -> None )
 
 let index_op_set_sugar (ident : Longident.t) args =
-  if
-    List.exists args ~f:(function Asttypes.Nolabel, _ -> false | _ -> true)
-    || List.is_empty args
-  then None
-  else
-    let args =
-      List.map args ~f:(function
-        | Asttypes.Nolabel, e -> e
-        | _ -> impossible "previous match" )
-    in
-    let e, args =
-      match List.rev args with
-      | e :: args -> (e, List.rev args)
-      | _ -> impossible "previous match"
-    in
+  match all_args_unlabeled args with
+  | None | Some [] -> None
+  | Some (e :: args) -> (
```

Comment:
`e` should be the last element of the list. Explaining the double List.rev above.
We could probably enumerate all three cases that we case about. from `[a1;e]`, `[a1;a2;e]` and `[a1;a2;a3;e]`

Reply:
Oops, my mistake. Enumerating the cases seems good to me. Probably doesn't matter, but likely faster too.